diff --git a/finders.c b/finders.c
--- a/finders.c
+++ b/finders.c
@@ -2461,6 +2461,6 @@ int genTower(PieceEnv *env, Piece *current, int depth)
                 continue;
             int brot = (rot + binfo[i][0]) & 3;
-            Piece *bridge = addEndCityPiece(env, base, brot,
+            Piece *bridge = addEndCityPiece(env, floor, brot,
                 binfo[i][1], binfo[i][2], binfo[i][3], BRIDGE_END);
             genPiecesRecusively(genBridge, env, bridge, depth+1);
         }
@@ -2501,6 +2501,6 @@ int genBridge(PieceEnv *env, Piece *current, int depth)
         int x = -8 + nextInt(env->rng, 8);
         int z = -70 + nextInt(env->rng, 10);
-        base = addEndCityPiece(env, base, rot, x, y, z, END_SHIP);
+        addEndCityPiece(env, base, rot, x, y, z, END_SHIP);
         *env->ship = 1;
     }
     else
