diff --git a/profotoce59/properties/VillageGenerator.java b/profotoce59/properties/VillageGenerator.java
--- a/profotoce59/properties/VillageGenerator.java
+++ b/profotoce59/properties/VillageGenerator.java
@@ -162,10 +162,10 @@
         Iterator<Piece> iterator = generationPieces.iterator();
         while(iterator.hasNext()) {
             Piece piece = iterator.next();
-            if(piece.type==PieceType.Feature){
-                confident = false;
-            }
             if (piece.box.intersects(chunkBox)){
+                if(piece.type==PieceType.Feature){
+                    confident = false;
+                }
                 if(!piece.place(rand, generator, confident, chunkBox)){
                     iterator.remove();
                 }
diff --git a/profotoce59/reecriture/VillagePools/PlainsVillageJigsawBlock.java b/profotoce59/reecriture/VillagePools/PlainsVillageJigsawBlock.java
--- a/profotoce59/reecriture/VillagePools/PlainsVillageJigsawBlock.java
+++ b/profotoce59/reecriture/VillagePools/PlainsVillageJigsawBlock.java
@@ -663,5 +663,5 @@
         ));
         this.put("common/well_bottom", Collections.singletonList(
-                new Pair<>(new Quad<>(PoolType.WELL_BOTTOM, "bottom", new Pair<>(BlockDirection.UP,BlockDirection.SOUTH), Blocks.COBBLESTONE), new BPos(4,3,4))
+                new Pair<>(new Quad<>(PoolType.WELL_BOTTOM, "bottom", new Pair<>(BlockDirection.UP,BlockDirection.NORTH), Blocks.COBBLESTONE), new BPos(3,2,0))
         ));
         this.put("oak",Collections.singletonList(
